File: arsenal/google-mcp/genai-toolbox/docs/en/getting-started/quickstart/js/adk/quickstart.js
import { InMemoryRunner, LlmAgent, LogLevel } from '@google/adk';
import { ToolboxClient } from '@toolbox-sdk/adk';

const prompt = `
You're a helpful hotel assistant. You handle hotel searching, booking, and
cancellations. When the user searches for a hotel, mention its name, id,
location and price tier. Always mention hotel ids while performing any
searches. This is very important for any operations. For any bookings or
cancellations, please provide the appropriate confirmation. Be sure to
update checkin or checkout dates if mentioned by the user.
Don't ask for confirmations from the user.
`;

const queries = [
  "Find hotels with Basel in its name.",
  "Can you book the Hilton Basel for me?",
  "Oh wait, this is too expensive. Please cancel it and book the Hyatt Regency instead.",
  "My check in dates would be from April 10, 2024 to April 19, 2024.",
];

process.env.GOOGLE_GENAI_API_KEY = process.env.GOOGLE_API_KEY || 'your-api-key'; // Replace it with your API key

export async function main() {
  const userId = 'test_user';
  const client = new ToolboxClient('http://127.0.0.1:5000');
  const tools = await client.loadToolset("my-toolset");

  const rootAgent = new LlmAgent({
    name: 'hotel_agent',
    model: 'gemini-2.5-flash',
    description: 'Agent for hotel bookings and administration.',
    instruction: prompt,
    tools: tools,
  });

  const appName = rootAgent.name;
  const runner = new InMemoryRunner({ agent: rootAgent, appName, logLevel: LogLevel.ERROR, });
  const session = await runner.sessionService.createSession({ appName, userId });

  for (const query of queries) {
    await runPrompt(runner, userId, session.id, query);
  }
}

async function runPrompt(runner, userId, sessionId, prompt) {
  const content = { role: 'user', parts: [{ text: prompt }] };
  const stream = runner.runAsync({ userId, sessionId, newMessage: content });
  const responses = await Array.fromAsync(stream);
  const accumulatedResponse = responses
    .flatMap((e) => e.content?.parts?.map((p) => p.text) ?? [])
    .join('');

  console.log(`\nMODEL RESPONSE: ${accumulatedResponse}\n`);
}

main();

File: arsenal/google-mcp/genai-toolbox/docs/en/getting-started/quickstart/python/adk/quickstart.py
# [START quickstart]
import asyncio

from google.adk import Agent
from google.adk.apps import App
from google.adk.runners import InMemoryRunner
from google.adk.tools.toolbox_toolset import ToolboxToolset
from google.genai.types import Content, Part

prompt = """
You're a helpful hotel assistant. You handle hotel searching, booking and
cancellations. When the user searches for a hotel, mention it's name, id,
location and price tier. Always mention hotel ids while performing any
searches. This is very important for any operations. For any bookings or
cancellations, please provide the appropriate confirmation. Be sure to
update checkin or checkout dates if mentioned by the user.
Don't ask for confirmations from the user.
"""

# TODO(developer): update the TOOLBOX_URL to your toolbox endpoint
toolset = ToolboxToolset(
    server_url="http://127.0.0.1:5000",
)

root_agent = Agent(
    name='hotel_assistant',
    model='gemini-2.5-flash',
    instruction=prompt,
    tools=[toolset],
)

app = App(root_agent=root_agent, name="my_agent")
# [END quickstart]

queries = [
    "Find hotels in Basel with Basel in its name.",
    "Can you book the Hilton Basel for me?",
    "Oh wait, this is too expensive. Please cancel it and book the Hyatt Regency instead.",
    "My check in dates would be from April 10, 2024 to April 19, 2024.",
]

async def main():
    runner = InMemoryRunner(app=app)
    session = await runner.session_service.create_session(
        app_name=app.name, user_id="test_user"
    )

    for query in queries:
        print(f"\nUser: {query}")
        user_message = Content(parts=[Part.from_text(text=query)])

        async for event in runner.run_async(user_id="test_user", session_id=session.id, new_message=user_message):
            if event.is_final_response() and event.content and event.content.parts:
                print(f"Agent: {event.content.parts[0].text}")

if __name__ == "__main__":
    asyncio.run(main())
